### ISSUE TITLE
Cancel Action

### ISSUE THREAD

**Issue by mordebip** [samtecspg/articulate#881]
### Feature Request

Slot Prompt count limit have resolved a lot of issues on the conversation flow, an other thing that i have seen on other chatbot framework and that could help articulate stand out is to have a cancel action.

Example 2 : 
- i'm hungry i want a pizza
- what topping do you want for your pizza
- Pepperoni and cheese please
- okay normal or double cheese
- you now what forget about it, i cancel this order
- ok your order have been canceled

Example 2 : 
- can you help me to reset my password
- yes, please provide me with your email address
- oh just cancel that, i found it

Basically it will exit the running flow and go through the Cancel Action.

**Comment by wrathagom** [comment]
You probably can already do this using an action + Webhooks + middleware to manually modify the session, but I don't have a working demo for you.

**Comment by wrathagom** [comment]
I think before too long we are going to merge modifiers into actions. Then we could expose additional actions like canceling or breaking out of flows.

**Comment by mordebip** [comment]
So basically i will have a cancel action that will call a middleware that will create a new sessionid ?

I was thinking about simething like rasa's /restart or some sort of solution that would drag the actuel Slot Prompt count limit to 0 for the running action.

**Comment by wrathagom** [comment]
Not a new session id, but rather modify the slot prompt as you say, or maybe remove that uncompleted action from the context? I can try to find time ti play with this, but it would likely be next week.
